### PR TITLE
Add compatibility with Espressif SDK 2.1.0 published in git repository

### PR DESCRIPTION
**Pull request by avr39-ripe**
Add compatibility with ESP SDK 2.1.0 by removing extra extern declaration of SDK stuff and by including proper SDK headers where needed

### CODE REVIEW

File: Sming/Libraries/Adafruit_ILI9341/Adafruit_ILI9341.cpp [hunk 1]
```diff
@@ -157,7 +157,8 @@ void Adafruit_ILI9341::begin(void) {
 	transmitCmdData(0xE1, data, 15);    	//Set Gamma
 
 	transmitCmd(0x11);    	//Exit Sleep
-	os_delay_us(120000);
```
Comment (slaff) [inline Sming/Libraries/Adafruit_ILI9341/Adafruit_ILI9341.cpp]: @avr39-ripe Can you remove that change from the PR. As far as I can see it has nothing to do with SDK 2.1.0.

**Comment by slaff** [comment]
@avr39-ripe Why not mandatory, it would be very helpful if we can add also SDK 2.1.0 in our CI system.

File: Sming/Libraries/Adafruit_ILI9341/Adafruit_ILI9341.cpp (hunk 1 above)
Comment (avr39-ripe) [inline Sming/Libraries/Adafruit_ILI9341/Adafruit_ILI9341.cpp]: it is mandatory change. without this this library do not compile any more. Old os_delay_us declaration was uint32_t and new, from esp sdk 2.1.0 is uint16_t, so I made such "strange" change here.

**Comment by avr39-ripe** [comment]
@slaff do not know nothing about how to add sdk 2.1.0 to CI :( can you help with it?

File: Sming/Libraries/Adafruit_ILI9341/Adafruit_ILI9341.cpp (hunk 1 above)
Comment (zgoda) [inline Sming/Libraries/Adafruit_ILI9341/Adafruit_ILI9341.cpp]: More code should be reviewed then to check if delays do not excede 65,535 limit.

**Comment by slaff** [comment]
Will be merged with rebase PR #1264

**Pull request closed without merging**